src/rulesets/LinearAlgebra/factorization.jl

```diff
@@ -1,6 +1,218 @@
+        if eltype(A) <: Real
+            ∂L = real(∂L)
+            ∂U = real(∂U)
+        end
+        ∂A = similar(factors)
+        m, n = size(A)
+        q = min(m, n)
+        if m == n  # square A
+            # ∂A = P' * (L' \ (tril(L' * ∂L, -1) + triu(∂U * U')) / U')
+            L = UnitLowerTriangular(factors)
+            U = UpperTriangular(factors)
+            ∂L isa AbstractZero ? fill!(∂A, 0) : mul!(∂A, L', ∂L)
+            if ∂U isa AbstractZero
+                fill!(UpperTriangular(∂A), 0)
+            else
+                copyto!(UpperTriangular(∂A), UpperTriangular(∂U * U'))
+            end
+            rdiv!(∂A, U')
+            ldiv!(L', ∂A)
+        elseif m < n  # wide A, system is [P*A1 P*A2] = [L*U1 L*U2]
+            @views begin
+                factors1 = factors[:, 1:q]
+                U2 = factors[:, (q + 1):n]
+                ∂A1 = ∂A[:, 1:q]
+                ∂A2 = ∂A[:, (q + 1):n]
+            end
+            L = UnitLowerTriangular(factors1)
+            U1 = UpperTriangular(factors1)
+            ∂U isa AbstractZero ? fill!(∂A, 0) : copyto!(∂A, ∂U)
+            triu!(rmul!(∂A1, U1'))
+            ∂tmp = ∂A2 * U2'
+            if ∂L isa AbstractZero
+                ∂A1 .-= tril!(∂tmp, -1)
+            else
+                ∂A1 .+= tril!(mul!(∂tmp, L', LowerTriangular(∂L), 1, -1), -1)
+            end
+            rdiv!(∂A1, U1')
+            ldiv!(L', ∂A)
+        else  # tall A, system is [P1*A; P2*A] = [L1*U; L2*U]
+            @views begin
+                factors1 = factors[1:q, :]
+                L2 = factors[(q + 1):m, :]
+                ∂A1 = ∂A[1:q, :]
+                ∂A2 = ∂A[(q + 1):m, :]
+            end
+            U = UpperTriangular(factors1)
+            L1 = UnitLowerTriangular(factors1)
+            ∂L isa AbstractZero ? fill!(∂A, 0) : copyto!(∂A, ∂L)
+            tril!(lmul!(L1', ∂A1), -1)
+            ∂tmp = L2' * ∂A2
+            if ∂U isa AbstractZero
+                ∂A1 .-= triu!(∂tmp)
+            else
+                ∂A1 .+= triu!(mul!(∂tmp, UpperTriangular(∂U), U', 1, -1))
+            end
+            ldiv!(L1', ∂A1)
+            rdiv!(∂A, U')
+        end
+        if pivot === Val(true)
+            ∂A = ∂A[invperm(F.p), :]
+        end
+        return NO_FIELDS, ∂A, DoesNotExist()
+    end
+    return F, lu_pullback
+end
+
+#####
+##### functions of `LU`
+#####
+
+# this rule standardizes the cotangent of LU to have property names :L and :U, which don't
+# have corresponding fields in LU.
+
+function rrule(::typeof(getproperty), F::TF, x::Symbol) where {T,TF<:LU{T,<:StridedMatrix{T}}}
+    function getproperty_LU_pullback(ΔY)
+        C = Composite{TF}
+        ∂F = if x === :L
+            C(; L=tril(ΔY, -1))
+        elseif x === :U
+            C(; U=triu(ΔY))
+        elseif x === :factors
+            m, n = size(F.factors)
+            q = min(m, n)
+            ∂L = tril(n == q ? ΔY : view(ΔY, :, 1:q), -1)
+            ∂U = triu(m == q ? ΔY : view(ΔY, 1:q, :))
```

Comment:
Slightly more constant folding friendly
```suggestion
            ∂L = tril(n === q ? ΔY : view(ΔY, :, 1:q), -1)
            ∂U = triu(m === q ? ΔY : view(ΔY, 1:q, :))
```